diff -uNr pdns-3.0/configure.ac pdns-3.0.new/configure.ac
--- pdns-3.0/configure.ac	2011-07-22 13:23:23.000000000 +0200
+++ pdns-3.0.new/configure.ac	2011-11-28 13:07:27.437416499 +0100
@@ -26,13 +26,13 @@
 
 dnl Check for lua
 AC_MSG_CHECKING(if with lua)
-AC_ARG_WITH(lua, AC_HELP_STRING([--with-lua],[lua]), WITH_LUA=$withval],[WITH_LUA=yes])
+AC_ARG_WITH(lua, AC_HELP_STRING([--with-lua],[lua]), [WITH_LUA=$withval],[WITH_LUA=yes])
 
 AC_MSG_RESULT($WITH_LUA)
 if test "$WITH_LUA" != "no"; then
  # try pkgconfig
  if test "$WITH_LUA" = "yes"; then
-      LUAPC=lua
+      LUAPC=lua51
  else
       LUAPC=$WITH_LUA
  fi
