From 2c010a9f0ff0e292d31b949065e1d5c245dc084e Mon Sep 17 00:00:00 2001
From: Morten Stevens <mstevens@fedoraproject.org>
Date: Mon, 21 Dec 2020 12:11:01 +0100
Subject: [PATCH] Fix missing #include for gcc-11

---
 pdns/lua-record.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pdns/lua-record.cc b/pdns/lua-record.cc
index 682759432ad..8d51c7937ed 100644
--- a/pdns/lua-record.cc
+++ b/pdns/lua-record.cc
@@ -1,3 +1,4 @@
+#include <thread>
 #include <future>
 #include <mutex>
 #include <boost/format.hpp>
