diff -uNr pdns-3.0/pdns/dnsparser.cc pdns-3.0.new/pdns/dnsparser.cc
--- pdns-3.0/pdns/dnsparser.cc	2011-07-22 13:23:22.000000000 +0200
+++ pdns-3.0.new/pdns/dnsparser.cc	2011-11-28 11:59:02.047547670 +0100
@@ -20,6 +20,7 @@
 #include "dnswriter.hh"
 #include <boost/lexical_cast.hpp>
 #include <boost/algorithm/string.hpp>
+#include <cstdio>
 
 #include "namespaces.hh"
 
diff -uNr pdns-3.0/pdns/rcpgenerator.cc pdns-3.0.new/pdns/rcpgenerator.cc
--- pdns-3.0/pdns/rcpgenerator.cc	2011-07-22 13:23:22.000000000 +0200
+++ pdns-3.0.new/pdns/rcpgenerator.cc	2011-11-28 11:59:22.501841684 +0100
@@ -22,6 +22,7 @@
 #include <boost/lexical_cast.hpp>
 #include <boost/algorithm/string.hpp>
 #include <iostream>
+#include <cstdio>
 #include "base32.hh"
 #include "base64.hh"
 #include "namespaces.hh"
