diff -uNr pdns-3.0/modules/ldapbackend/ldapbackend.cc pdns-3.0.new/modules/ldapbackend/ldapbackend.cc
--- pdns-3.0/modules/ldapbackend/ldapbackend.cc	2011-07-22 13:23:22.000000000 +0200
+++ pdns-3.0.new/modules/ldapbackend/ldapbackend.cc	2011-11-28 11:55:30.704283338 +0100
@@ -434,7 +434,7 @@
         							continue;
         						}
 
-        						rr.priority = (uint16_t) strtoul( (content.substr( 0, first )).c_str(), &endptr, 10 );
+        						rr.priority = (u_int16_t) strtoul( (content.substr( 0, first )).c_str(), &endptr, 10 );
         						if( *endptr != '\0' )
         						{
         							L << Logger::Warning << m_myname << " Invalid " << attrname << " without priority for " << m_qname << ": " << content << endl;
diff -uNr pdns-3.0/modules/ldapbackend/powerldap.cc pdns-3.0.new/modules/ldapbackend/powerldap.cc
--- pdns-3.0/modules/ldapbackend/powerldap.cc	2011-07-22 13:23:22.000000000 +0200
+++ pdns-3.0.new/modules/ldapbackend/powerldap.cc	2011-11-28 11:55:46.355019073 +0100
@@ -4,7 +4,7 @@
 
 
 
-PowerLDAP::PowerLDAP( const string& hosts, uint16_t port, bool tls )
+PowerLDAP::PowerLDAP( const string& hosts, u_int16_t port, bool tls )
 {
         d_ld = 0;
         d_hosts = hosts;
diff -uNr pdns-3.0/modules/ldapbackend/powerldap.hh pdns-3.0.new/modules/ldapbackend/powerldap.hh
--- pdns-3.0/modules/ldapbackend/powerldap.hh	2011-07-22 13:23:22.000000000 +0200
+++ pdns-3.0.new/modules/ldapbackend/powerldap.hh	2011-11-28 11:54:08.347077606 +0100
@@ -18,7 +18,7 @@
  */
 
 
-
+#include <sys/types.h>
 #include <map>
 #include <string>
 #include <vector>
@@ -69,7 +69,7 @@
         typedef map<string, vector<string> > sentry_t;
         typedef vector<sentry_t> sresult_t;
 
-        PowerLDAP( const string& hosts = "ldap://127.0.0.1/", uint16_t port = LDAP_PORT, bool tls = false );
+        PowerLDAP( const string& hosts = "ldap://127.0.0.1/", u_int16_t port = LDAP_PORT, bool tls = false );
         ~PowerLDAP();
 
         void getOption( int option, int* value );
