--- trunk/pdns/pdns/logger.cc	(revision 304)
+++ trunk/pdns/pdns/logger.cc	(revision 305)
@@ -1,11 +1,10 @@
 /*
     PowerDNS Versatile Database Driven Nameserver
-    Copyright (C) 2002  PowerDNS.COM BV
+    Copyright (C) 2005  PowerDNS.COM BV
 
     This program is free software; you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
-    the Free Software Foundation; either version 2 of the License, or
-    (at your option) any later version.
+    it under the terms of the GNU General Public License version 2 as 
+    published by the Free Software Foundation
 
     This program is distributed in the hope that it will be useful,
     but WITHOUT ANY WARRANTY; without even the implied warranty of
@@ -131,6 +130,16 @@
   return *this;
 }
 
+Logger& Logger::operator<<(unsigned long i)
+{
+  ostringstream tmp;
+  tmp<<i;
+
+  *this<<tmp.str();
+
+  return *this;
+}
+
 
 Logger& Logger::operator<<(ostream & (&)(ostream &))
 {
--- trunk/pdns/pdns/logger.hh	(revision 304)
+++ trunk/pdns/pdns/logger.hh	(revision 305)
@@ -1,11 +1,10 @@
 /*
     PowerDNS Versatile Database Driven Nameserver
-    Copyright (C) 2002  PowerDNS.COM BV
+    Copyright (C) 2005  PowerDNS.COM BV
 
     This program is free software; you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
-    the Free Software Foundation; either version 2 of the License, or
-    (at your option) any later version.
+    it under the terms of the GNU General Public License version 2 as 
+    published by the Free Software Foundation
 
     This program is distributed in the hope that it will be useful,
     but WITHOUT ANY WARRANTY; without even the implied warranty of
@@ -111,6 +110,7 @@
   Logger& operator<<(const string &s);   //!< log a string
   Logger& operator<<(int);   //!< log an int
   Logger& operator<<(unsigned int);   //!< log an unsigned int
+  Logger& operator<<(unsigned long);   //!< log an unsigned int
   Logger& operator<<(Urgency);    //!< set the urgency, << style
 
 #ifndef WIN32
