--- pdns-2.9.17/modules/ldapbackend/utils.hh	2004-02-22 10:35:01.000000000 +0100
+++ pdns-2.9.17.new/modules/ldapbackend/utils.hh	2005-05-20 11:15:40.530121464 +0200
@@ -73,7 +73,7 @@
 }


-inline string ip2ptr4( string ip )
+inline string ip2ptr4( const string& ip )
 {
 	string ptr;
 	vector<string> parts;
@@ -89,7 +89,7 @@
 }


-inline string ip2ptr6( string ip )
+inline string ip2ptr6( const string& ip )
 {
 	string ptr, part, defstr;
 	vector<string> parts;
@@ -116,4 +116,19 @@
 	return ptr + "ip6.arpa";
 }

+
+inline string strbind( const string& search, const string& replace, string subject )
+{
+	size_t pos = 0;
+
+
+	while( ( pos = subject.find( search, pos ) ) != string::npos )
+	{
+		subject.replace( pos, search.size(), replace );
+		pos += replace.size();
+	}
+
+	return subject;
+}
+
 #endif
