diff -urN orig/configure my/configure
--- orig/configure	2004-02-28 21:39:56.000000000 +0100
+++ my/configure	2004-03-11 12:35:50.000000000 +0100
@@ -21470,7 +21470,7 @@
 	PGSQL_libdir=
 	for m in $PGSQL_lib_check; do
 	        if test -d "$m" && \
-		   (test -f "$m/libpq.a" || test -f "$m/libpq++.a")
+		   (test -f "$m/libpq.so" || test -f "$m/libpq++.so")
 	        then
 	                PGSQL_libdir=$m
 	                break
