diff -urN org/pdns/communicator.hh pdns-2.9.13/pdns/communicator.hh
--- org/pdns/communicator.hh	Sat May 24 20:39:31 2003
+++ pdns-2.9.13/pdns/communicator.hh	Thu Jan 15 02:16:00 2004
@@ -94,7 +94,7 @@
   
   time_t earliest()
   {
-    time_t early=1<<31-1; // y2038 problem lurking here :-)
+    time_t early=(1<<31)-1; // y2038 problem lurking here :-)
     for(d_nqueue_t::const_iterator i=d_nqueue.begin();i!=d_nqueue.end();++i) 
       early=min(early,i->next);
     return early-time(0);
