--- pdns-2.9.22/modules/ldapbackend/powerldap.hh~	2009-02-07 19:46:56.713653002 +0100
+++ pdns-2.9.22/modules/ldapbackend/powerldap.hh	2009-02-07 19:47:40.993231763 +0100
@@ -18,7 +18,7 @@
  */
 
 
-
+#include <sys/types.h>
 #include <map>
 #include <string>
 #include <vector>
diff -Nur pdns-2.9.21.1.orig/modules/ldapbackend/ldapbackend.cc pdns-2.9.21.1.int16/modules/ldapbackend/ldapbackend.cc
--- pdns-2.9.21.1.orig/modules/ldapbackend/ldapbackend.cc	2007-04-21 07:56:36.000000000 -0600
+++ pdns-2.9.21.1.int16/modules/ldapbackend/ldapbackend.cc	2008-10-06 10:08:55.000000000 -0600
@@ -433,7 +433,7 @@
 								continue;
 							}
 
-							rr.priority = (uint16_t) strtoul( (content.substr( 0, first )).c_str(), &endptr, 10 );
+							rr.priority = (u_int16_t) strtoul( (content.substr( 0, first )).c_str(), &endptr, 10 );
 							if( *endptr != '\0' )
 							{
 								L << Logger::Warning << m_myname << " Invalid " << attrname << " without priority for " << m_qname << ": " << content << endl;
diff -Nur pdns-2.9.21.1.orig/modules/ldapbackend/powerldap.cc pdns-2.9.21.1.int16/modules/ldapbackend/powerldap.cc
--- pdns-2.9.21.1.orig/modules/ldapbackend/powerldap.cc	2007-04-21 07:56:36.000000000 -0600
+++ pdns-2.9.21.1.int16/modules/ldapbackend/powerldap.cc	2008-10-06 10:08:55.000000000 -0600
@@ -2,7 +2,7 @@
 
 
 
-PowerLDAP::PowerLDAP( const string& hosts, uint16_t port, bool tls )
+PowerLDAP::PowerLDAP( const string& hosts, u_int16_t port, bool tls )
 {
 	int protocol = LDAP_VERSION3;
 
diff -Nur pdns-2.9.21.1.orig/modules/ldapbackend/powerldap.hh pdns-2.9.21.1.int16/modules/ldapbackend/powerldap.hh
--- pdns-2.9.21.1.orig/modules/ldapbackend/powerldap.hh	2007-04-21 07:56:36.000000000 -0600
+++ pdns-2.9.21.1.int16/modules/ldapbackend/powerldap.hh	2008-10-06 10:11:53.000000000 -0600
@@ -78,7 +74,7 @@
 	typedef map<string, vector<string> > sentry_t;
 	typedef vector<sentry_t> sresult_t;
 
-	PowerLDAP( const string& hosts = "ldap://127.0.0.1/", uint16_t port = LDAP_PORT, bool tls = false );
+	PowerLDAP( const string& hosts = "ldap://127.0.0.1/", u_int16_t port = LDAP_PORT, bool tls = false );
 	~PowerLDAP();
 
 	void getOption( int option, int* value );
