diff -uNr pdns-3.0/modules/ldapbackend/powerldap.hh pdns-3.0.new/modules/ldapbackend/powerldap.hh
--- pdns-3.0/modules/ldapbackend/powerldap.hh	2011-07-22 13:23:22.000000000 +0200
+++ pdns-3.0.new/modules/ldapbackend/powerldap.hh	2011-11-28 11:50:02.192161656 +0100
@@ -27,6 +27,8 @@
 #include <inttypes.h>
 #include <errno.h>
 #include <lber.h>
+
+#define LDAP_DEPRECATED 1
 #include <ldap.h>
 
 #ifdef HAVE_CONFIG_H
