--- trunk/pdns/pdns/logger.cc	(revision 306)
+++ trunk/pdns/pdns/logger.cc	(revision 307)
@@ -140,6 +140,15 @@
   return *this;
 }
 
+Logger& Logger::operator<<(long i)
+{
+  ostringstream tmp;
+  tmp<<i;
+
+  *this<<tmp.str();
+
+  return *this;
+}
 
 Logger& Logger::operator<<(ostream & (&)(ostream &))
 {
--- trunk/pdns/pdns/logger.hh	(revision 306)
+++ trunk/pdns/pdns/logger.hh	(revision 307)
@@ -110,6 +110,7 @@
   Logger& operator<<(const string &s);   //!< log a string
   Logger& operator<<(int);   //!< log an int
   Logger& operator<<(unsigned int);   //!< log an unsigned int
+  Logger& operator<<(long);   //!< log an unsigned int
   Logger& operator<<(unsigned long);   //!< log an unsigned int
   Logger& operator<<(Urgency);    //!< set the urgency, << style
 
